addon/serializer.js

```diff
@@ -271,52 +271,43 @@ class Serializer {
   _maybeAddAssociationIds(model, attrs) {
     let newHash = _assign({}, attrs);
 
-    if (this.serializeIds === 'always') {
-      model.associationKeys.forEach((key) => {
-        let association = model[key];
+    let associationKeys = this._associationKeysToInclude(model);
+
+    associationKeys.forEach((key) => {
+      let association = model[key];
+
+      if (model.associationFor(key).isPolymorphic) {
+        if (this.isCollection(association)) {
+          let formattedKey = this.keyForRelationshipIds(key);
+          newHash[formattedKey] = model[`${singularize(key)}Ids`];
+        } else {
+          let formattedTypeKey = this.keyForPolymorphicForeignKeyType(key);
+          let formattedIdKey = this.keyForPolymorphicForeignKeyId(key);
+
+          newHash[formattedTypeKey] = model[`${key}Id`].type;
+          newHash[formattedIdKey] = model[`${key}Id`].id;
+        }
+      } else {
         if (this.isCollection(association)) {
           let formattedKey = this.keyForRelationshipIds(key);
           newHash[formattedKey] = model[key].models.map((m) => m.id);
         } else if (association) {
           let formattedKey = this.keyForForeignKey(key);
           newHash[formattedKey] = model[`${key}Id`];
         }
-      });
-    } else if (this.serializeIds === 'included') {
-      this.getKeysForIncluded().forEach((key) => {
-        let association = model[key];
+      }
+    });
 
-        if (model.associationFor(key).isPolymorphic) {
-          if (association instanceof PolymorphicCollection) {
-            let formattedKey = this.keyForRelationship(key);
-
-            newHash[formattedKey] = model[`${singularize(key)}Ids`];
-          } else if (association instanceof Collection) {
-            let formattedKey = this.keyForRelationshipIds(key);
```

Comment:
Could this code be reached? All tests stayed green, when I removed this case.

Reply:
I'm really uncomfortable deleting this much code, even if the tests go green.

Reply:
Keeping aside the problem that the branch for polymorphic relations seemed to be missing completely from the `this.serializeIds === 'always'`case, I also couldn't really see why the cases `this.serializeIds === 'always'` and `this.serializeIds === 'included'` should be handled differently at all, when it comes to how the ids are being serialized.

As far as the deleted code is concerned, I didn't really remove that much logic. Lines 277-283 of the original code were duplicates of lines 306-313. The actual bits of logic that were altered are:
* the branch for polymorphic relations is now also present in the `this.serializeIds === 'always' case
* in the _polymorphic_ case `PolymorphicCollection` and `Collection` are treated the same (this is admittedly the bit I'm most uncertain of, too)
 * instead of calling the `keyForRelationship` hook (line 291 of old code), the `keyForRelationshipIds` hook is now called. For the reasoning behind this, please see the second point here https://github.com/samselikoff/ember-cli-mirage/issues/1274#issuecomment-371300993